### PR TITLE
chore: Use helper resolver for tags

### PR DESCRIPTION
**Pull request by candiduslynx**
#### Summary

Added a dedicated resolver for tags (`[]types.Tag` -> `map[string]string`).
Additionally, supporting passing structure, containing `Tags` field as well.

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

**Review by roneli — APPROVED** [review]
LGTM, i'd wait for @disq review this as he was the initial implementor of this code.

**Review by amanenk — APPROVED** [review]
LGTM

**Review by amanenk — APPROVED** [review]
Nice job. Good idea of moving custom types to a separate file. a few nits with comments.

File: resources/services/lightsail/alarms.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func Alarms() *schema.Table {
 		IgnoreError:   client.IgnoreAccessDeniedServiceDisabled,
 		DeleteFilter:  client.DeleteAccountRegionFilter,
 		Options:       schema.TableCreationOptions{PrimaryKeys: []string{"arn"}},
-		IgnoreInTests: true, // it can't be included into e2e test, because there is no way to deploy it using terraform
+		IgnoreInTests: true,
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply resources/services/lightsail/alarms.go:23]: Unfortunately, the next run of cq-gen will just remove it.
Possibly, we can add this support, if HCL parsing supports comments as well.

File: resources/services/lightsail/instance_snapshots.go [hunk 2]
```diff
@@ -20,8 +20,8 @@ func InstanceSnapshots() *schema.Table {
 		Multiplex:     client.ServiceAccountRegionMultiplexer("lightsail"),
 		IgnoreError:   client.IgnoreAccessDeniedServiceDisabled,
 		DeleteFilter:  client.DeleteAccountRegionFilter,
-		IgnoreInTests: true, // can't be deployed using terraform.
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply resources/services/lightsail/instance_snapshots.go:23]: cq-gen

**Pull request merged**